js/styles/_typography.scss

```diff
@@ -13,10 +41,148 @@ $govuk-typography-use-rem: false;
   @include govuk-responsive-margin(4, "bottom");
 }
 
+.govuk-heading-m {
+  @extend %govuk-heading-m;
+}
+
+%govuk-heading-s {
+  @include govuk-text-colour;
+  @include govuk-font($size: 19, $weight: bold);
+
+  display: block;
+
+  margin-top: 0;
+  @include govuk-responsive-margin(4, "bottom");
+}
+
+.govuk-heading-s {
+  @extend %govuk-heading-s;
+}
+
+// Captions to be used inside headings
+
+.govuk-caption-xl {
+  @include govuk-font($size: 27);
+
+  display: block;
+
+  margin-bottom: govuk-spacing(1);
+
+  color: $govuk-secondary-text-colour;
+}
+
+.govuk-caption-l {
+  @include govuk-font($size: 24);
+
+  display: block;
+
+  margin-bottom: govuk-spacing(1);
+  color: $govuk-secondary-text-colour;
+
+  @include govuk-media-query($from: tablet) {
+    margin-bottom: 0;
+  }
+}
+
+.govuk-caption-m {
+  @include govuk-font($size: 19);
+
+  display: block;
+
+  color: $govuk-secondary-text-colour;
+}
+
+// Body (paragraphs)
+
+%govuk-body-l {
+  @include govuk-text-colour;
+  @include govuk-font($size: 24);
+
+  margin-top: 0;
+  @include govuk-responsive-margin(6, "bottom");
+}
+
+.govuk-body-l {
+  @extend %govuk-body-l;
+}
+
 %govuk-body-m {
```

Comment:
I think before we just had the % because it doesn't result in any output - so a class selector like `.govuk-body-m` will result in some css output, however a placeholder selector `%govuk-body-m` will be available to be extended by other stuff but by itself not result in any output. I guess we define it in here, maybe because govuk has the classes and placeholders in the same class so by importing one we got extra classes being output. Where we just wanted to be able to:
```scss
.cbanner-govuk-heading-m {
  @extend %govuk-heading-m;
}
```
in govuk-frontend.scss without impacting other styles on the page / having more css to inject that we need by just @import'ing it inside a class to avoid pollution

Reply:
I kind of wonder if it's still the case we have to do this or if sass in govuk-frontend has been refactored a bit - also seems like it would be a good idea to have some tests that say "we don't redefine styles for govuk classes" so do a grep on the output css to ensure there is no `.govuk-*` in there

Reply:
Is there anything you want to change / add now, related to the above comments?

Reply:
we should remove all the extra styles in here, it should just be % prefixed bits- and only the heading-m and body-m

if we don't we'll end up changing styles on the page (possibly)

Reply:
we could also remove the
```
@import "node_modules/govuk-frontend/dist/govuk/base";
```
because we import it in the govuk-frontend.scss above this file